### PR TITLE
[openrr-apps] Support plugin

### LINKED ISSUE #312: Support openrr-plugin in openrr-apps
follow-up #304

### PR DESCRIPTION
**Pull request by taiki-e**
Closes #312

### Examples

```sh
# build plugin
cargo build -p openrr-plugin-example

# run app
cargo run -p openrr-apps --bin openrr_apps_joint_position_sender -- \
  --config-path ./openrr-apps/config/sample_robot_client_config_for_plugin.toml
```

### CODE REVIEW

File: openrr-apps/src/robot_config.rs [hunk 1]
```diff
@@ -322,54 +666,149 @@ impl RobotConfig {
...
         };
-        Ok(raw_joint_trajectory_clients)
+
+        #[cfg(feature = "ros")]
+        clients.extend(arci_ros::create_joint_trajectory_clients_lazy(
+            ros_clients_configs,
+            urdf_robot.as_ref(),
+        )?);
+
+        for (plugin_name, config) in &self.plugins {
```
Comment (OTL) [inline openrr-apps/src/robot_config.rs:776]: I feel it's better to make this a function, like  `create_plugin_xxx_yyy` to reduce the lines and make it easy to understand.

Comment (OTL) [reply openrr-apps/src/robot_config.rs:776]: Around the code, L678 has a similar to this. It's nice if these are merged **if possible**. (I think it's not easy)

File: openrr-apps/src/robot_config.rs [hunk 2]
```diff
@@ -322,54 +666,149 @@ impl RobotConfig {
...
+                    return Err(Error::DuplicateInstance(format!(
+                        "Multiple {:?} instances {:?} are found. Consider renaming one of the instances",
+                        instance.type_, instance.name,
+                    )));
+                }
+                let plugin = plugins.load(plugin_name)?.unwrap();
+                let args = instance.load_args()?;
+                let plugin_name = plugin_name.clone();
+                let instance_name = instance.name.clone();
+                let instance_kind = instance.type_;
```
Comment (OTL) [inline openrr-apps/src/robot_config.rs:794]: This is repeated again and again. How about creating a function for this?

Comment (taiki-e) [reply openrr-apps/src/robot_config.rs]: done in https://github.com/openrr/openrr/pull/319/commits/4bb644a3649f6bf891b91438827b6336530b0a54

**Review by OTL — APPROVED** [review]

**Pull request merged**